server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -1462,6 +1463,9 @@ public IndexOptions parseIndexOptions(String fieldName, Map<String, ?> indexOpti
                 RescoreVector rescoreVector = null;
                 if (indexVersion.onOrAfter(ADD_RESCORE_PARAMS_TO_QUANTIZED_VECTORS)) {
                     rescoreVector = RescoreVector.fromIndexOptions(indexOptionsMap, indexVersion);
+                    if (rescoreVector == null) {
```

Comment:
This should only happen on new indices. Please add a new index version to bar changing this on existing indices.